internal/throttler/threshold/controller.go

```diff
@@ -0,0 +1,54 @@
+package threshold
+
+import (
+	"context"
+
+	"github.com/openfga/openfga/internal/throttler"
+)
+
+type dispatchThrottlingThresholdType uint32
+
+const (
+	dispatchThrottlingThreshold dispatchThrottlingThresholdType = iota
+)
+
+type Config struct {
+	Enabled      bool
+	Throttler    throttler.Throttler
+	Threshold    uint32
+	MaxThreshold uint32
+}
+
+func ShouldThrottle(ctx context.Context, currentCount uint32, defaultThreshold uint32, maxThreshold uint32) bool {
+	threshold := defaultThreshold
+
+	if maxThreshold == 0 {
+		maxThreshold = defaultThreshold
+	}
+
+	thresholdInCtx := ThrottlingThresholdFromContext(ctx)
+
+	if thresholdInCtx > 0 {
+		threshold = min(thresholdInCtx, maxThreshold)
+	}
+
+	return currentCount > threshold
+}
+
+// ContextWithThrottlingThreshold will save the dispatch throttling threshold in context.
+func ContextWithThrottlingThreshold(ctx context.Context, threshold uint32) context.Context {
```

Comment:
Can we keep this in pkg/telemetry/context.go instead of here?  The reason is that if other software uses this as library and they want to set their throttling value, they will not have access to internal directory.

Reply:
Ah! That's why it was inside `pkg/telemetry`...

I was the one to move it to an internal pkg because i didn't know it'd be needed outside. Please document that :)

`pkg/telemetry` is the wrong place because this is not about telemetry at all. Let's create a new package for this

Reply:
I have fixed this.